src/vdom/component.js

```diff
@@ -20,11 +20,13 @@ export function setComponentProps(component, props, opts, context, mountAll) {
 	if ((component.__ref = props.ref)) delete props.ref;
 	if ((component.__key = props.key)) delete props.key;
 
-	if (!component.base || mountAll) {
-		if (component.componentWillMount) component.componentWillMount();
-	}
-	else if (component.componentWillReceiveProps) {
-		component.componentWillReceiveProps(props, context);
+	if (typeof component.constructor.getDerivedStateFromProps === 'undefined') {
```

Comment:
I'd be open to loosening this check if it can help offset the size.

Reply:
By loosening do you mean making the check less specific (i.e `if (!component.constructor.getDerivedStateFromProps)`)? I just tried that change in commit 3f51408 and it actually increased the size by 2 bytes.

Reply:
@andrewiggins I think he meant removing the `if`-statement completely.

Reply:
Oh, whoops. That makes sense. Yea, that would save about 4 bytes I believe.